crates/levm/src/vm.rs

```diff
@@ -47,3 +88,51 @@ impl VM {
         self.pc
     }
 }
+
+#[derive(Debug, Clone, Default)]
+pub struct Memory {
+    data: Vec<u8>,
+}
+
+impl Memory {
+    pub fn new() -> Self {
+        Self { data: Vec::new() }
+    }
+
+    fn resize(&mut self, offset: usize) {
+        if (offset + 32).next_multiple_of(32) > self.data.len() {
+            self.data.resize((offset + 32).next_multiple_of(32), 0);
```

Comment:
This assumes that the memory needs to be resized to hold up to `offset + 32` bytes. This can be wrong on `MSTORE8` or `MCOPY`.

Reply:
That's right, for example in the `mcopy` test it was resizing to 128 when it had to resize to 96. Thanks!